nostrpostrlib/src/main/java/nostr/postr/Utils.kt

```diff
@@ -81,4 +81,11 @@ fun Int.toByteArray(): ByteArray {
 }
 
 fun ByteArray.toNsec() = Bech32.encodeBytes(hrp = "nsec", this, Bech32.Encoding.Bech32)
-fun ByteArray.toNpub() = Bech32.encodeBytes(hrp = "npub", this, Bech32.Encoding.Bech32)
\ No newline at end of file
+fun ByteArray.toNpub() = Bech32.encodeBytes(hrp = "npub", this, Bech32.Encoding.Bech32)
+
+fun String.rawHexForm() = Bech32.decodeBytes(this).second
+
+fun String.rawHexFormWithHrp(): Pair<String, ByteArray> {
+    val decodedHex = Bech32.decodeBytes(this)
+    return Pair(decodedHex.first, decodedHex.second)
+}
```

Comment:
Nothing hex-y here neither. With some Kotlin magic I get this one-liner:

```suggestion
/**
 * Interpret the string as Bech32 encoded and return hrp and ByteArray as Pair
 **/
fun String.bechToHrpBytes() = Bech32.decodeBytes(this).run { Pair(first, second) }
```

Reply:
That's true! I forgot about this.:sweat_smile: